core/services/pipeline/runner.go

```diff
@@ -350,7 +351,10 @@ func (r *runner) run(
 				continue
 			}
 			errors = append(errors, result.Error)
-			outputs = append(outputs, result.Output.Val)
+
+			if result.Output.Val != nil {
```

Comment:
I was considering something like this when changing the pipeline but I think some code had the expectation that there would be one output for each task (and one error), with the ones that error being set to nil (some code indexes Output[task.OutputIndex()] or something like that).

I'd be up for removing the nils but we need to check nothing breaks because of this and that it's backwards compatible.

Reply:
Yea, there are some issues on FE side so I added this "hack". I'll remove that and will fix FE side.